cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -4742,8 +4751,8 @@ function(build_orc)
         "-DSNAPPY_HOME=${ORC_SNAPPY_ROOT}"
         "-DSNAPPY_LIBRARY=$<TARGET_FILE:${Snappy_TARGET}>"
         "-DLZ4_HOME=${ORC_LZ4_ROOT}"
-        "-DLZ4_LIBRARY=$<TARGET_FILE:LZ4::lz4>"
-        "-DLZ4_STATIC_LIB=$<TARGET_FILE:LZ4::lz4>"
+        "-DLZ4_LIBRARY=$<LZ4_TARGET>"
+        "-DLZ4_STATIC_LIB=$<LZ4_TARGET>"
```

Comment:
This + the `set(LZ4_TARGET lz4_static)` were an attempt to resolve the issue in the crossbow jobs, but it definitely doesn't work:

```
CMake Error at /usr/share/cmake-3.28/Modules/ExternalProject.cmake:2179 (file):
  Error evaluating generator expression:

    $<TARGET_FILE:LZ4::lz4>

  Target "LZ4::lz4" is not an executable or library.
```

Reply:
Sure! I tried this but now get:

```
CMake Error in c++/src/CMakeLists.txt:
  Imported target "orc_lz4" includes non-existent path

    "/build/cpp/orc_ep-prefix/src/orc_ep-build/$<BUILD_INTERFACE:/build/cpp/_deps/lz4_ep-src/include"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

  * The path was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and references files it does not
  provide.
```

Reply:
Thanks for all of your help on this!

Reply:
I've pushed thee fix of this. Could you check it?